contrib_bots/lib/github_detail.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import print_function
+
+import re
+import os
+import sys
+import logging
+import six.moves.configparser
+
+import requests
+
+class GithubHandler(object):
+    '''
+    This bot provides details on github issues and pull requests when they're
+    refrenced in the chat.
+    '''
+
+    config_path = os.path.expanduser('~/.contrib_bots/github_detail.ini')
+    GITHUB_ISSUE_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{repo}/issues/{id}'
+    GITHUB_PULL_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{reepo}/pulls/{id}'
+
+    def __init__(self):
+        config = six.moves.configparser.ConfigParser()
+        config.readfp(open(self.config_path))
+        if config.get('Github', 'owner'):
+            self.owner = config.get('Github', 'owner')
+        else:
+            # Allowing undefined default repos would require multiple triage_message regexs.
+            # It's simpler to require them to be defined.
+            sys.exit('Default owner not defined')
+
+        if config.get('Github', 'repo'):
+            self.repo = config.get('Github', 'repo')
+        else:
+            sys.exit('Default repo not defined')
+
+    def usage(self):
+        # type: () -> None
+        return ("This plugin displays details on github issues and pull requests. "
+                "To refrence an issue or pr type @github-detail then anytime in the message it's id like '#2561'."
+                "If you want to refrence an issues from a repository other than the default use"
+                "the user/repo#id form.")
```

Comment:
Done, the usage instructions aren't perfect but I don't want the message to be too long. I think what I have now should be a fine explanation, considering most users of this bot will be developers.